Eslint refactor

ap0nia opened the pull request:
Simplified eslint config and cleaned up some files while I was at it

ap0nia committed:
initial commit for linting refactor

ap0nia committed:
holy moly

ap0nia committed:
ok fixed

ap0nia committed:
fixed scripts in package json

ap0nia committed:
added comma back to tsconfig

ap0nia commented on src/index.tsx:
```diff
@@ -1,8 +1,3 @@
 import { render } from 'react-dom';
-
 import App from './App';
-import { unregister } from './registerServiceWorker';
```
I'm pretty sure this is purely an artifact of CRA, and not relevant in Vite. Removal of `registerServiceWorker` altogether should be fine

EricPedley replied:
Yeah, confirmed with network inspector that removing this doesn't stop the page from caching itself.

ap0nia commented on src/components/RightPane/SectionTable/SectionTableBody.tsx:
```diff
@@ -95,7 +96,8 @@ const CourseCodeCell = withStyles(styles)((props: CourseCodeCellProps) => {
         <NoPaddingTableCell className={classes.cell}>
             <Tooltip title="Click to copy course code" placement="bottom" enterDelay={300}>
-                <div
+                <Button
```
Section code in the right course pane is now a button; doesn't seem to significantly impact the UI. But I can style it more to make it seem "invisible" like before

ap0nia commented on src/components/RightPane/CoursePane/SearchForm/AdvancedSearch.tsx:
```diff
@@ -250,18 +251,19 @@ class AdvancedSearch extends PureComponent<AdvancedSearchProps, AdvancedSearchSt
         return (
             <>
-                <div onClick={this.handleExpand} className={classes.container}>
+                <Button
```
Section code in the calendar popup is now a button; doesn't seem to significantly impact the UI. But I can style it more to make it seem "invisible" like before.

EricPedley replied:
It's fine no need to fix the styling.

ap0nia committed:
can remove eslint ignore

ap0nia commented on src/components/Calendar/CourseCalendarEvent.tsx:
```diff
@@ -145,18 +145,19 @@ const CourseCalendarEvent = (props: CourseCalendarEventProps) => {
                         <tr>
                             <td className={classes.alignToTop}>Section code</td>
                             <Tooltip title="Click to copy course code" placement="right">
-                                {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
```
This can be removed since onClick was moved into the button with the section code

ap0nia committed:
revert lintstagedrc

EricPedley commented on .eslintrc.yml:
```diff
@@ -1,54 +1,35 @@
+    - react
+    - react-hooks
     - import
-    - simple-import-sort
```
I kinda liked having auto import sorting. Are there any other isort plugins that we can get that don't screw up the leaflet imports? It's fine if not.

ap0nia replied:
Sure, I'll turn on this rule. https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md

ap0nia committed:
turned on eslint import/order rule

ap0nia committed:
ran the import/order linter

- if your imports are out of order, it will throw an error
- running the lint command with the `--fix` flag will automatically fix
  the order, basically the same as the import order plugin
- `yarn lint --fix`

EricPedley reviewed (approved):
lgtm
